### ISSUE TITLE
torch_geometric.utils.softmax()

### ISSUE THREAD

**Issue by yrf1** [pyg-team/pytorch_geometric#723]
## ❓ Questions & Help

I am getting a Segmentation fault when calling `torch_geometric.utils.softmax(src, idx, num_nodes)`.

I fed in inputs like this:
- src is torch.Size([1266, 4])
- idx is torch.Size([1266])
- num_nodes is 561

Can someone tell me what went wrong please? Or another alternative way to call a similar function so that the code can run? Thank you!

**Comment by rusty1s** [comment]
Do you get a segmentation fault when running, *e.g.*, `examples/gat.py`? If not, this may indicate that your idx tensor contains invalid entries greater than or equal to 561.

**Comment by yrf1** [comment]
Hi I got the segmentation fault when running examples/gat.py. I've been trying different combinations of package versions to run a code I found on Github that specifies running on torch_geometric==1.0.3.

Is there an alternative to the torch_geometric.utils.softmax() that I can stick in place for the code? Or can you tell me which combination of package versions in which torch-geometric 1.0.3 would work? Currently I'm using for ex:
- torch 1.0.0
- torch-cluster 1.2.4
- torch-geometric 1.0.3
- torch-scatter 1.3.1
- torch-sparse 0.2.4
- torchvision 0.2.2
- CUDA 80
- CUDNN 4.0
- gcc 4.8.5

Or any other useful advice would be greatly appreciated.

Thank you!

**Comment by rusty1s** [comment]
Hi, for `torch-scatter`, you need to use the last version pre-PyTorch 1.1, which is actually [v1.1.2](https://github.com/rusty1s/pytorch_scatter/releases). For PyTorch Geometric, you could try out [v1.3.0](https://github.com/rusty1s/pytorch_geometric/releases). To verify that your installation actually works, you can manually clone `torch-scatter`, checkout version 1.1.2, install it and run the test suite.

**Comment by yrf1** [comment]
Is there any other option if I need to run some github repo that specifies only working on torch-geometric 1.0.3?

**Comment by rusty1s** [comment]
You can also use `torch-geometric==1.0.3` with `torch-scatter==1.1.2`. The API didn't change.

**Comment by KnowingNothing** [comment]
I have the same problem. I am using torch-geometric==1.3.2 with torch-scatter==1.1.2. I find that by commenting off the scatter_max in softmax function I can get rid of the segmentation fault. BTW, why do you use the scatter_max? I guess you want to avoid number overflow caused by exp() function.

**Comment by rusty1s** [comment]
That is exactly the reason. Commenting this out is not recommended, since the segmentation fault suggests that the installation of torch-scatter failed.

**Comment by KnowingNothing** [comment]
I think you are right, by correctly installing torch-scatter, I can correctly run softmax.

**Comment by twjiang** [comment]
I have the same Segmentation fault problem when calling function scatter_max() in softmax(). What do you mean by "correctly installing torch-scatter"? 

The following is what I have installed, is there any problem?
pip install torch_geometric==1.4.1
pip install torch_sparse==0.4.4
pip install torch_scatter==1.4.0
pip install torch_cluster==1.4.5

**Comment by rusty1s** [comment]
This just means that there occurred an error during installation of the CUDA kernels that prevents a successful execution. How did you try to install `torch-scatter`?

**Comment by twjiang** [comment]
I just used pip command to install torch_scatter, like: pip install torch_scatter==1.4.0, and it just worked.
